modules/test/playwright/tests/roles-admin-web/accountRoles.spec.ts

```diff
@@ -0,0 +1,161 @@
+	accountRolesPage,
+	accountsPage,
+	apiHelpers,
+	page,
+}) => {
+	const userAccount = await apiHelpers.headlessAdminUser.postUserAccount();
+
+	userData[userAccount.alternateName] = {
+		name: userAccount.givenName,
+		password: 'test',
+		surname: userAccount.familyName,
+	};
+
+	const companyId = await page.evaluate(() => {
+		return Liferay.ThemeDisplay.getCompanyId();
+	});
+
+	const accountAdminRole = await apiHelpers.headlessAdminUser.postRole({
+		name: getRandomString(),
+		rolePermissions: [
+			{
+				actionIds: ['ACCESS_IN_CONTROL_PANEL'],
+				primaryKey: companyId,
+				resourceName:
+					'com_liferay_account_admin_web_internal_portlet_AccountEntriesAdminPortlet',
+				scope: 1,
+			},
+		],
+	});
+
+	await apiHelpers.headlessAdminUser.assignUserToRole(
+		accountAdminRole.externalReferenceCode,
+		userAccount.id
+	);
+
+	const account1 = await apiHelpers.headlessAdminUser.postAccount({
+		name: 'Account1',
+		type: 'business',
+	});
+
+	apiHelpers.data.push({id: account1.id, type: 'account'});
+
+	const account2 = await apiHelpers.headlessAdminUser.postAccount({
+		name: 'Account2',
+		type: 'business',
+	});
+
+	apiHelpers.data.push({id: account2.id, type: 'account'});
+
+	await apiHelpers.headlessAdminUser.assignUserToAccountByEmailAddress(
+		account1.id,
+		[userAccount.emailAddress]
+	);
+	await apiHelpers.headlessAdminUser.assignUserToAccountByEmailAddress(
+		account2.id,
+		[userAccount.emailAddress]
+	);
+
+	const accountRole1 = await apiHelpers.headlessAdminUser.postRole({
+		name: getRandomString(),
+		rolePermissions: [
+			{
+				actionIds: ['UPDATE', 'VIEW_ACCOUNT_ROLES'],
+				primaryKey: '0',
+				resourceName: 'com.liferay.account.model.AccountEntry',
+				scope: 3,
+			},
+			{
+				actionIds: ['VIEW'],
+				primaryKey: '0',
+				resourceName: 'com.liferay.account.model.AccountRole',
+				scope: 3,
+			},
+		],
+		roleType: 'account',
+	});
+
+	let rolesResponse = await apiHelpers.headlessAdminUser.getAccountRoles(
+		account1.id
+	);
+
+	let role = rolesResponse?.items?.filter(
+		(role) => role.name === accountRole1.name
+	);
+
+	await apiHelpers.headlessAdminUser.assignUserToAccountRole(
+		account1.id,
+		role[0].id,
+		userAccount.id
+	);
+
+	const accountRole2 = await apiHelpers.headlessAdminUser.postRole({
+		name: getRandomString(),
+		rolePermissions: [
+			{
+				actionIds: ['UPDATE'],
+				primaryKey: '0',
+				resourceName: 'com.liferay.account.model.AccountEntry',
+				scope: 3,
+			},
+		],
+		roleType: 'account',
+	});
+
+	rolesResponse = await apiHelpers.headlessAdminUser.getAccountRoles(
+		account2.id
+	);
+
+	role = rolesResponse?.items?.filter(
+		(role) => role.name === accountRole2.name
+	);
+
+	await apiHelpers.headlessAdminUser.assignUserToAccountRole(
+		account2.id,
+		role[0].id,
+		userAccount.id
+	);
+
+	await performLogout(page);
+	await performLogin(page, userAccount.alternateName);
+
+	await page.goto(`${PORTLET_URLS.accountAdmin}`);
```

Comment:
usually we create a method goTo with this in the page (example `SiteSettingsPage` or `JournalStructuresPage`. Can we do the same here (I think in the accountsPage)?

Reply:
@smottal there reason for this is that there is already a goto() method in accountsPage which will not work here because for some reason we cannot access accountsAdminPortlet via applicationMenu. Probably some permission issue but I think it's irrelevant for this test, so I used the direct URL. I could create a method like gotoURL or something like that...

Reply:
Alternatively, I could refactor the goto method and see if it can reach the URL via menu if not then use direct URL.

Reply:
so I did this as an alternative:
```
async goto() {
		try {
			await Promise.race([
				this.applicationsMenuPage.goToAccounts(),
				new Promise((_, reject) =>
					setTimeout(() => reject('Timeout'), 4000)
				),
			]);
		}
		catch {
			await this.page.goto(`${PORTLET_URLS.accountAdmin}`);
		}
	}
```